src/main/java/Duke.java

```diff
@@ -1,10 +1,181 @@
-public class Duke {
+import duke.DukeException;
+import duke.Parser;
+import duke.Storage;
+import duke.Ui;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.stage.Stage;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.layout.Region;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+
+import tasks.TaskList;
+
+import java.io.IOException;
+
+/**
+ * The Duke program implements a chatbot called Duke that
+ * supports queries such as creating, marking and deleting tasks.
+ *
+ * @author Chen Hsiao Ting
+ * @version 1.0
+ * @since 2021-08-13
+ */
+
+public class Duke extends Application {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.loadTask());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        } catch (IOException e) {
+            System.out.println("ERROR: " + e.getMessage());
+        }
+    }
+
+    public Duke() { }
+
+    /**
+     * Runs the Duke chatbot program until user input the bye command.
+     */
+    public void run() {
+        Ui.welcome();
+        Parser.startDuke(tasks);
+        Storage.saveList();
+    }
+
+    /**
+     * Starts the Duke chatbot program
+     *
+     * @param args input path for the data file
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("data/duke.txt").run();
+    }
+
+    @Override
+    public void start(Stage stage) {
```

Comment:
Some recently committed methods should have a Javadoc header comment if they're public